From: Uwe Kleine-König <ukleinek@debian.org>
Subject: usbip: Use _FORTIFY_SOURCE=2 instead of (implicitly) =1
Date: Wed, 03 May 2023 19:36:22 +0200
Forwarded: https://lore.kernel.org/linux-usb/20230503173622.1072787-1-ukleinek@debian.org/

_FORTIFY_SOURCE=2 uses more and stricter checks. This is what e.g.
Debian recommends to build packages with.

While at it fix a typo in the output of ./configure --help.

Signed-off-by: Uwe Kleine-König <ukleinek@debian.org>
Reviewed-By: Hongren Zheng <i@zenithal.me>
---
Changes since (implicit) v1:

- s/_FORTIFY_SROUCE/_FORTIFY_SOURCE/ in the help text, noted (off-list)
  by Cyril Brulebois.

 tools/usb/usbip/configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/usb/usbip/configure.ac b/tools/usb/usbip/configure.ac
index 607d05c5ccfd..8debf934f8b7 100644
--- a/tools/usb/usbip/configure.ac
+++ b/tools/usb/usbip/configure.ac
@@ -94,11 +94,11 @@ AC_SUBST([USBIDS_DIR])
 AC_MSG_CHECKING([whether to use fortify])
 AC_ARG_WITH([fortify],
 	    [AS_HELP_STRING([--with-fortify],
-			    [use _FORTIFY_SROUCE option when compiling)])],
+			    [use _FORTIFY_SOURCE=2 option when compiling)])],
 			    dnl [ACTION-IF-GIVEN]
 			    [if test "$withval" = "yes"; then
 				AC_MSG_RESULT([yes])
-				CFLAGS="$CFLAGS -D_FORTIFY_SOURCE -O"
+				CFLAGS="$CFLAGS -D_FORTIFY_SOURCE=2 -O"
 			     else
 			     	AC_MSG_RESULT([no])
 				CFLAGS="$CFLAGS -U_FORTIFY_SOURCE"
-- 
2.39.2
